src/test/java/com/aws/greengrass/clientdevices/auth/connectivity/ConnectivityInformationTest.java

```diff
@@ -0,0 +1,200 @@
+        doReturn(getConnectivityInfoResponse).when(greengrassV2DataClient)
+                .getConnectivityInfo(any(GetConnectivityInfoRequest.class));
+
+        List<ConnectivityInfo> connectivityInfos = connectivityInformation.getConnectivityInfo();
+        verify(greengrassV2DataClient, times(1))
+                .getConnectivityInfo(any(GetConnectivityInfoRequest.class));
+        assertThat(connectivityInfos, containsInAnyOrder(connectivityInfo, connectivityInfo1));
+    }
+
+    @Test
+    void GIVEN_no_connectivity_info_WHEN_get_connectivity_info_THEN_no_connectivity_info_returned() {
+        GetConnectivityInfoResponse getConnectivityInfoResponse = GetConnectivityInfoResponse.builder().build();
+        doReturn(getConnectivityInfoResponse).when(greengrassV2DataClient)
+                .getConnectivityInfo(any(GetConnectivityInfoRequest.class));
+
+        List<ConnectivityInfo> connectivityInfos = connectivityInformation.getConnectivityInfo();
+        verify(greengrassV2DataClient, times(1))
+                .getConnectivityInfo(any(GetConnectivityInfoRequest.class));
+        assertThat(connectivityInfos, is(empty()));
+    }
+
+    @Test
+    void GIVEN_cloudThrowValidationException_WHEN_get_connectivity_info_THEN_no_connectivity_info_returned(
+            ExtensionContext context) {
+        ignoreExceptionOfType(context, ValidationException.class);
+        when(greengrassV2DataClient.getConnectivityInfo(any(GetConnectivityInfoRequest.class)))
+                .thenThrow(ValidationException.class);
+
+        assertThat(connectivityInformation.getConnectivityInfo(), is(empty()));
+    }
+
+    @Test
+    void GIVEN_cached_connectivity_info_WHEN_get_cached_connectivity_info_THEN_connectivity_info_returned() {
+        ConnectivityInfo connectivityInfo = ConnectivityInfo.builder().hostAddress("172.8.8.10")
+                .metadata("").id("172.8.8.10").portNumber(8883).build();
+        ConnectivityInfo connectivityInfo1 = ConnectivityInfo.builder().hostAddress("localhost")
+                .metadata("").id("localhost").portNumber(8883).build();
+        GetConnectivityInfoResponse getConnectivityInfoResponse = GetConnectivityInfoResponse.builder()
+                .connectivityInfo(Arrays.asList(connectivityInfo, connectivityInfo1)).build();
+        doReturn(getConnectivityInfoResponse).when(greengrassV2DataClient)
+                .getConnectivityInfo(any(GetConnectivityInfoRequest.class));
+
+        connectivityInformation.getConnectivityInfo();
+        List<String> connectivityInfos = connectivityInformation.getCachedHostAddresses();
+        assertThat(connectivityInfos, containsInAnyOrder("172.8.8.10", "localhost"));
+    }
+
+    @Test
+    void GIVEN_missingConnectivityInfo_WHEN_getConnectivityInformation_THEN_returnEmptySet() {
+        Set<HostAddress> connectivityInfo = connectivityInformation.getAggregatedConnectivityInformation();
```

Comment:
Suggestion: Test against useCases and assert the output, that way we don't have to change the test if the implementation of the useCase changes - This is fine regardless as a unit test but I believe integration tests triumph unit tests